R/conversion.R

```diff
@@ -159,20 +159,7 @@ py_to_r.datetime.datetime <- function(x) {
 
 #' @export
 r_to_py.Date <- function(x, convert = FALSE) {
-
-  datetime <- import("datetime", convert = FALSE)
-  items <- lapply(x, function(item) {
-    iso <- strsplit(format(item), "-", fixed = TRUE)[[1]]
-    year <- as.integer(iso[[1]])
-    month <- as.integer(iso[[2]])
-    day <- as.integer(iso[[3]])
-    datetime$date(year, month, day)
-  })
-
-  if (length(items) == 1)
-    items[[1]]
-  else
-    r_to_py_impl(items, convert)
+  r_convert_date(x, convert)
```

Comment:
This may need to handle length-one lists still (to ensure they're returned as scalars)

Reply:
Not sure exactly what you mean? I was trying to find what this
https://github.com/rstudio/reticulate/blob/64f685ab279516250ef15f2957820744c9841d32/R/conversion.R#L172
was for but didn't manage to get into the `else` branch...

Reply:
E.g.

```
> library(reticulate)
> r_to_py(Sys.Date())
2019-11-15
> r_to_py(c(Sys.Date(), Sys.Date()))
[datetime.date(2019, 11, 15), datetime.date(2019, 11, 15)]
```

That is, we make a Python list for R vectors of length > 1, and a 'scalar' for R vectors of length 1. Is this behavior still preserved in this PR?

Reply:
Oh, got it, thanks! Does this look better now?